CONTRIBUTING.md

```diff
@@ -243,3 +243,29 @@ error_session:
 	return NULL;
 }
 ```
+
+## Bisecting
+
+When bisecting, it is often necessary to build older commits with their
+contemporary version of wlroots:
+
+```python
+#!/usr/bin/env python3
+#
+# Check out the wlroots commit that was the current master at the time that the current
+# sway HEAD was committed. Requires pygit2. E.g.: apk add py3-pygit2
+
+import pygit2
+
+sway = pygit2.Repository(".")
+head = sway.revparse_single("HEAD")
+sway_time = head.commit_time
+
+wlroots = pygit2.Repository("subprojects/wlroots")
+master = wlroots.revparse_single("master")
+for commit in wlroots.walk(master.id, pygit2.enums.SortMode.TIME):
```

Comment:
I'm not sure if using the time is always the right thing here, have you encountered any issue with that so far?

Reply:
> I'm not sure if using the time is always the right thing here

In this context, `TIME` is the _commit_ time specifically, which is the time when the commit was merged into the master branch.

The time shown by `git log` is the _author_ time. Using that one does not work; it yields bogus results since some commits are authored months before being merged.

> have you encountered any issue with that so far?

When wlroots version N is released, the version is immediately bumped to N+1. From that point on, meson will complain that it has wlroots N+1 but expects version N.

In my experience bisecting sway, very few commits have this issue. Usually I skip these. If you need to dive into that range, you'll can just check out the wlroots tag.